org.jacoco.core.test/src-kotlin/org/jacoco/core/test/validation/targets/KotlinDataClassTarget.kt

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Fabian Mastenbroek - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.test.validation.targets
+
+import org.junit.Assert.assertEquals
+
+/**
+ * This test target creates a Kotlin data class.
+ */
+class KotlinDataClassTarget {
+
+    internal data class DataClass(var value: String) // $line-data-class$
+
+    companion object {
+        @JvmStatic
+        fun main(args: Array<String>) {
```

Comment:
Maybe this can be simplified by moving `main` to top-level, so that `companion object` and `@JvmStatic` can be removed?

Reply:
If you move `main` to top-level, you need to reference to it from Java with the `KotlinDataClassTargetKt`. This does not work since the test setup code (using `Source.getSourceFor`)  will then look for a `KotlinDataClassTargetKt.kt` source file which obviously does not exist.

Reply:
`Source.getSourceFor` can be adapted 😉

Reply:
How would you approach it? Checking for a `Kt` suffix in the class name and changing the file name accordingly?

Reply:
One can imagine reading of `SourceFile` attribute from class file, but we don't do this for Java classes using KISS principle :wink: Thus yep, just check of `Kt` suffix looks fine to me.